src/components/PopupWithForm.js

```diff
@@ -22,6 +23,14 @@ export default class PopupWithForm extends Popup {
     return inputsItems;
   }
 
+  renderLoading(loading) {
+    if (loading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
+    }
+  }
+
   setEventListeners() {
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
```

Comment:
Needs correction (line 38): please don't close the popup here, close it only after the successful api response in `then` block in index.js

Reply:
Corrected!